optuna/storages/_rdb/storage.py

```diff
@@ -590,135 +578,136 @@ def _update_trial(
+                trial_param = (
+                    session.query(models.TrialParamModel)
+                    .filter(models.TrialParamModel.trial_id == trial_id)
+                    .all()
+                )
+                trial_param_dict = {attr.param_name: attr for attr in trial_param}
+                for name, v in params.items():
+                    if name in trial_param_dict:
+                        trial_param_dict[
+                            name
+                        ].distribution_json = distributions.distribution_to_json(
+                            distributions_[name]
+                        )
+                        trial_param_dict[name].param_value = v
+                        session.add(trial_param_dict[name])
+                trial_model.params.extend(
+                    models.TrialParamModel(
+                        param_name=param_name,
+                        param_value=param_value,
+                        distribution_json=distributions.distribution_to_json(
+                            distributions_[param_name]
+                        ),
                     )
-                    trial_param_dict[name].param_value = v
-                    session.add(trial_param_dict[name])
-            trial_model.params.extend(
-                models.TrialParamModel(
-                    param_name=param_name,
-                    param_value=param_value,
-                    distribution_json=distributions.distribution_to_json(
-                        distributions_[param_name]
-                    ),
+                    for param_name, param_value in params.items()
+                    if param_name not in trial_param_dict
                 )
-                for param_name, param_value in params.items()
-                if param_name not in trial_param_dict
-            )
-        session.add(trial_model)
-        self._commit(session)
+            session.add(trial_model)
 
         return True
 
     def set_trial_state(self, trial_id: int, state: TrialState) -> bool:
 
-        session = self.scoped_session()
-
-        trial = models.TrialModel.find_by_id(trial_id, session, for_update=True)
-        if trial is None:
-            session.rollback()
-            raise KeyError(models.NOT_FOUND_MSG)
-
-        self.check_trial_is_updatable(trial_id, trial.state)
-
-        if state == TrialState.RUNNING and trial.state != TrialState.WAITING:
-            session.rollback()
+        try:
+            with _scoped_session(self.scoped_session) as session:
+                trial = models.TrialModel.find_by_id(trial_id, session, for_update=True)
+                if trial is None:
+                    session.rollback()
+                    raise KeyError(models.NOT_FOUND_MSG)
+
+                self.check_trial_is_updatable(trial_id, trial.state)
+
+                if state == TrialState.RUNNING and trial.state != TrialState.WAITING:
+                    session.rollback()
```

Comment:
Ditto?
```suggestion
```

Reply:
Since there are no insert or update before this line, I think I can remove this line.